Update ExpectedException analyzer message/description to use the new Assert.Throws methods (and fix typo)

Youssef1313 opened the pull request.

Evangelink reviewed (commented):
Looks like you are changing more than what the PR title says. Please fix content or title.

Youssef1313 commented:
The other extra changes were mostly already done in AreNotSame. They unintentionally ended up in this branch. Rebasing made them go away, and also adjusted the PR title as I also fixed a typo in the messages I'm modifying.

Youssef1313 commented:
/azp run
